### ISSUE TITLE
launch_template should not allow vpc_security_groups if network_interfaces are defined

### ISSUE THREAD

**Issue by DeltaByte** [terraform-providers/terraform-provider-aws#14440]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.12.28
+ provider.aws v3.0.0
+ provider.cloudflare v2.7.0
+ provider.tls v2.1.1
```

### Affected Resource(s)

* aws_launch_template

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "example" {
  name_prefix   = "ecs-api"
  instance_type = "t3a.micro"
  image_id      = "example-ami"

  iam_instance_profile {
    arn = aws_iam_instance_profile.ecs.arn
  }

  vpc_security_group_ids = [
    aws_security_group.example.id
  ]

  network_interfaces {
    associate_public_ip_address = true
  }
}
```

### Expected Behavior

Terraform should have prevented the `vpc_security_gorup_ids` being set when there is a `network_interfaces` block present.

### Actual Behavior

Terraform dpeloyed the launch template with no issues.

### Important Factoids

I am unsure if this is something that would be caught by the EC2 APIs when using the launch_template for something else,
but in my case, I am using it for an `aws_spot_fleet_request`, while the launch_configuration seems fine at first glance, as soon
as my Spot Fleet tries to actually provision a new instance it errors out.

```
Error message: Network interfaces and an instance-level security groups may not be specified on the same request (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameterCombination; Proxy: null)
```

**Comment by dfitz82** [comment]
Terraform version: 0.13.5
aws v3.24.1

Plans are created successfully.  I am creating the launch template and ASG in the same file.
The Launch Template is created.  Through the console I can see the network interface is configured with a security group.
I receive the following error during the apply.

```
Error: Error creating Auto Scaling Group: InvalidQueryParameter: Invalid launch template: When a network interface is provided, the security groups must be a part of it.
	status code: 400, request id: 10203e5a-83f4-4be4-8c7a-2cd8583e8c87
```
Additional Information:  
Originally I had created the LT with option 'associate_public_ip_address=false'.  I added the 'network_interfaces' block to add public IPs to my instances and received the error above inside of the ASG Activity log.  I was able to use terraform to add the security group to the network interface and create a new version of the LT.
When I adjusted capacity on the ASG it was stuck at "Updating Capacity".  I realize why this happened, but I wanted to provide more information in case it helps.

**Comment by justinretzolk** [comment]
Hi all 👋 Thank you for reporting this. Given that there have been a few Terraform and AWS provider releases between when this issue was filed and now, are you able to confirm if this is still an issue you're encountering?

**Comment by eschulma** [comment]
Yes, I just ran into this exact same thing in 5.12. And now I will move the security group into `network_interfaces`.

**Comment by JamJar00** [comment]
This is still an issue on v5.31.0 when you use an invalid launch template with an Auto Scaling Group. Only worked out what it was because I eventually stumbled on this issue
